src-electron/rest/file-ops.js

```diff
@@ -41,24 +41,31 @@ const studio = require('../ide-integration/studio-rest-api')
 function httpPostFileOpen(db) {
   return async (req, res) => {
     let { zapFilePath, ideProjectPath } = req.body
-    let name = ''
+    if (req.body.search?.includes('filePath=%2F')) {
```

Comment:
The ideProjectPath is passed in from the Studio side already. 
Is there any specific reason why it needs to be embedded into "search" and why is it called "search"?

Reply:
search is an array object within "Window.Location" which was already storing objects. I just added a new one to conform with what was already there.

Reply:
to further explain, I completely changed how ZAP opens a file. It no longer uses the API renderer.

Reply:
@tecimovic the "search" object has many different parameters. In order to find the file I had to parse the entire string. @jingteng25742 found a better way to do it which I will make a new PR and fix.